packages/docusaurus-theme-classic/src/validateThemeConfig.js

```diff
@@ -250,8 +250,8 @@ const ThemeConfigSchema = Joi.object({
   announcementBar: Joi.object({
     id: Joi.string().default('announcement-bar'),
     content: Joi.string(),
-    backgroundColor: Joi.string().default('#fff'),
-    textColor: Joi.string().default('#000'),
+    backgroundColor: Joi.string(),
```

Comment:
Initially, `backgroundColor` and `textColor` were added in the announcement bar to override the default values. So we definitely shouldn't define them in the scheme.

Reply:
This would be a breaking change in behavior for many sites that expect black text on a white banner.

![image](https://user-images.githubusercontent.com/749374/121924515-45571980-cd3c-11eb-9a87-c607d0db9512.png)

I don't think changing to primary color would be better in most cases. Some users will complain about that change, and some will only notice it months after due to not seeing the banner anymore on their own website (due to having dismissed it).

If we really want this, we'd rather do that on another PR, explaining the motivations and highlight it as a breaking change made on purpose. I'd prefer to keep it as it is today

Reply:
I see, but now announcement bar has bad support in dark mode. That is why I initially used primary color as background, because then this color will be most suitable for both color modes.

Reply:
> I see, but now announcement bar has bad support in dark mode. 

Not sure to understand how the current PR improves dark mode support. Both modes looks the same to me, it's just the background color that is different.

In any case I'd rather have a distinct PR for this change as it's not a minor refactor without any impact like the rest of the PR

Reply:
Because white background color shouldn't be used in dark mode at all. And as a rule, the primary color will never be white or black, so it is most suitable for both modes. But okay, won't argue, and just replaced the background color with white in the CSS file itself.